packages/pds/src/api/com/atproto/sync/listHostedRepos.ts

```diff
@@ -0,0 +1,61 @@
+import { InvalidRequestError } from '@atproto/xrpc-server'
+import { Server } from '../../../../lexicon'
+import AppContext from '../../../../context'
+import { Cursor, GenericKeyset, paginate } from '../../../../db/pagination'
+
+export default function (server: Server, ctx: AppContext) {
+  server.com.atproto.sync.listHostedRepos(async ({ params }) => {
+    const { limit, cursor } = params
+    const ref = ctx.db.db.dynamic.ref
+    const innerBuilder = ctx.db.db
+      .selectFrom('repo_root')
+      .innerJoin('user_account', 'user_account.did', 'repo_root.did')
+      .where('repo_root.takedownId', 'is', null)
+      .select([
+        'repo_root.did as did',
+        'repo_root.root as head',
+        'user_account.createdAt as createdAt',
+      ])
+    let builder = ctx.db.db.selectFrom(innerBuilder.as('repos')).selectAll()
+    const keyset = new TimeDidKeyset(ref('createdAt'), ref('did'))
```

Comment:
We probably don't need to have a wrapping `select *` here, believe it should just involve qualifying the keyset columns, e.g. `createdAt -> user_account.createdAt` and `did -> user_account.did`.

Reply:
If there's ever an index placed on `user_account` to support this pagination, we may want to check whether selecting from `user_account` and joining over to `repo_root` is smoother for the db.  I am pretty sure postgres figures this out, though, as long as both sort columns are on the same table.